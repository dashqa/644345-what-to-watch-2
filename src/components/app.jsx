import React from "react";
import MainPage from "./main-page/main-page.jsx";

const App = () => {
  const films = [`Fantastic Beasts`, `Bohemian Rhapsody`, `Macbeth`, `Spider Man`];
  return <MainPage
    films={films}
  />;
};

export default App;
