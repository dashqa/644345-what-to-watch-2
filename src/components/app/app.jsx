import React from "react";
import {Switch, Route, Router} from "react-router-dom";
import history from "@store/history";
import {Fetch} from "@constants";

import SignIn from "@pages/sign-in/sign-in";
import MainPage from "@pages/main-page/main-page";
import MovieDetails from "@pages/movie-details/movie-details";
import MoviePlayer from "@pages/movie-player/movie-player";
import AddReview from "@pages/add-review/add-review";
import MyList from "@pages/my-list/my-list";
import PageNotFound from "@pages/page-not-found/page-not-found";

import withLoaded from "@hocs/with-loaded/with-loaded";

const App = () => {
  return (
    <Router history={history}>
      <Switch>
        <Route exact path='/' component={MainPage}/>
        <Route exact path='/films/:id' component={MovieDetails}/>
        <Route path='/films/:id/review' component={AddReview}/>
        <Route path='/films/:id/player' component={MoviePlayer}/>
        <Route path='/my-list' component={MyList}/>
        <Route path='/login' component={SignIn}/>
        <Route component={PageNotFound}/>
      </Switch>
    </Router>
  );
};

export {App};

export default withLoaded(Fetch.MOVIES, Fetch.PROMO)(App);
