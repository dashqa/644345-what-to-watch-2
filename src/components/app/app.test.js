import React from 'react';
import {shallow} from "enzyme";
import toJSON from "enzyme-to-json";
import {App} from "./app";

it(`App correctly renders after relaunch`, () => {
  const tree = shallow(<App/>);
  expect(toJSON(tree)).toMatchSnapshot();
});
