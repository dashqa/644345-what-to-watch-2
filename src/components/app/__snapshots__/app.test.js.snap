// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(WithLoading)",
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(WithLoading)",
            "type": [Function],
          }
        }
        path="/films/:id"
      />
    </Switch>
  </BrowserRouter>
</Provider>
`;
